### ISSUE TITLE
How to open specific page

### ISSUE THREAD

**Issue by manzajaro** [phonegap-build/PushPlugin#213]
Hello, I would like to know how can I open a specific page in the application by clicking on the notification 

Thanks

**Comment by Neotrixsx** [comment]
Thanks oscarabilleira Really You Save My Day.  Your Answer Work Perfectly Work For my Appplication.

**Comment by briankostar** [comment]
which 'onresume' event are you talking about? the one in PushHandlerActivity.java?

**Comment by oscarabilleira** [comment]
Onresume is an event used in phonegap, just add:

document.addEventListener("resume", onResume, false);

and a function onReume where I put the code that you want to execute during "onresume" event.

 function onResume() {

;}

To change the view I just "read" an extra-data sent by push notification and save it using localStorage.setItem... then inside "onresume" function and also during the "normal starup" I read this extra-value using localStorage.getItem... and acording its value the app change the view... if this extra-value is null the app starts in default page.

**Comment by briankostar** [comment]
Thank you :)

**Comment by tejal123** [comment]
hey,
i'm very much new to the android development domain, can u plz explain me in which files do we have to make the changes??as well as where do we find the extra id??

**Comment by lumierephoto** [comment]
@oscarabilleira Do you think you can share your onResume function code?

**Comment by oscarabilleira** [comment]
As you can see in the code that I posted before the, "id" value  (extra value sent with the app) iis saved using  window.localStorage.setItem("push_que", e.payload.id);  and     window.localStorage.setItem("push_que", e.id);  (for android and ios, because in wp8 I can't make it work)

After that in my "onResume function, acording the value of the id one or other page is loaded.... 
For example:
function onResume() {
var que_push=window.localStorage.getItem("que_push");
if (que_push=="news") {$.mobile.changePage("#news")  ;}
if (que_push=="vote") {$.mobile.changePage("#pool") ; }
// if a number is sent  open a specific news article calling a specific funcion that loads the "articles"
if (que_push!="vote" and que_push!="vote" and que_push|=""} 
{
load_news(que_push);
;}

;}



I'ts just an example I hope it can help you :)

**Comment by lumierephoto** [comment]
Thanks for your quick reply!

It seems you're using jQuery Mobile/jQuery for changing page, but if I was using "plain" Javascript could I do something like this?

    function onResume() {
      var que_push=window.localStorage.getItem("que_push");
         if (que_push=="news") { window.location.href = "page.html";          window.localStorage.removeItem("push_que") ;}
         if (que_push=="vote") { window.location.href = "anotherpage.html";    window.localStorage.removeItem("push_que"); }
        // if a number is sent  open a specific news article calling a specific funcion that loads the "articles"
          if (que_push!="vote" and que_push!="vote" and que_push!=""} 
     {
          window.localStorage.removeItem("push_que")
          load_news(que_push);
      ;}
    ;}

Sorry, my Javascript chops aren't that great.

**Comment by oscarabilleira** [comment]
Of course... in fact initally I use window.location.href instead of changepage, but using jquery works better hehe

**Comment by lumierephoto** [comment]
@oscarabilleira   May I ask where onResume() is triggered in your app?

**Comment by oscarabilleira** [comment]
In my app the onresume function is in  index.html (my app has only one html file), I can put it in any place of it.

But   document.addEventListener("resume", onResume, false); is inside   onDeviceReady() function

**Comment by lumierephoto** [comment]
OK, so you're calling it with this line inside the onDeviceReady() :

     document.addEventListener("resume", onResume, false);

**Comment by oscarabilleira** [comment]
Yes that's right

**Comment by lumierephoto** [comment]
Awesome! Thanks.

**Comment by akashposin** [comment]
Thanks, How can i come to know that if app is opened by notification or opened by tapping from background apps?

**Comment by ramyacl** [comment]
Can any one please suggest this onResume function will call while receiving notification?I need to increment badges while receiving notifications from server.

**Comment by Kunze** [comment]
What about you receive 5 notifications? this code will save only the last

**Comment by rahulbussa** [comment]
In case of cold start of App,neither resume event nor the onNotification() method is getting invoked in my case.On clicking notification only app is getting launched.Some one please help us .I am using this in Ionic.

**Comment by harakhovich** [comment]
@rahulbussa hey! Have you solved your issue?

**Comment by rahulbussa** [comment]
@Maxyss after little bit search i figured out.
I had put my $cordovaPush:notificationReceived outside of onDeviceReady() event  in app.js ,
this solved the issue,now the listener is getting executed even on cold start.

**Comment by Aniket-ajency** [comment]
Hey Guys this is bit late to ask this question but I have the same issue as @rahulbussa was having and I have written my $cordovaPush:notificationReceived out side my device ready but still I am not getting any trigger when my app is minimized please help me out.

**Comment by rahulbussa** [comment]
@Aniket-ajency  are you getting notification on foreground?

**Comment by Aniket-ajency** [comment]
yes I am. But when my app is in coldstart I am able to get notification but cordova event is not getting triggered.can you please help me out with this I am stuck in this problem from 5 days.

**Comment by rahulbussa** [comment]
@Aniket-ajency  ,for that issue i had moved my cordova listner code in app.js and tied up it with $rootScope.FInd below sample code,and place the listner out of onDeviceReady event.

```
.run(function($ionicPlatform,$rootScope,$cordovaPush,$cordovaDialogs){
   $rootScope.$on('$cordovaPush:notificationReceived', function (event, notification) {
     if (ionic.Platform.isAndroid()) {
         handleAndroid(notification);
     }
```

  });

   function handleAndroid(notification) {
              var customerprofile = '';
              console.log("In foreground " + notification.foreground  + " Coldstart " + notification.coldstart);
              if (notification.event == "registered") {
                  var regId = notification.regid;
                      //Logic to store device token in back end.
                    $cordovaDialogs.alert(notification.message, "Push Notification Received");
                  }
             else if (notification.event == "message") {
                console.log(notification);
                //Code to handle Once Notification is received
                $cordovaDialogs.alert(notification.message, "Push Notification Received");
                  }
            else if (notification.event == "error")
                $cordovaDialogs.alert(notification.msg, "Push notification error event");
           else
               $cordovaDialogs.alert(notification.event, "Push notification handler - Unprocessed Event");
            }

//Registering for push notification
$ionicPlatform.ready(function() {

```
  document.addEventListener("deviceready", onDeviceReady, false);


  function onDeviceReady() {
        var config = {

                        "senderID": "PUT YOUR SENDER ID",
                       "ecb": "angular.element(document.body).injector().get('$cordovaPush').onNotification"
                        };

                       $cordovaPush.register(config).then(function (result) {
                          console.log("Register success " + result);
                          $cordovaToast.showShortCenter('Registered for push notifications');
                          });

  }
```

}

Hope this code helps you.

**Comment by Aniket-ajency** [comment]
@rahulbussa thanks for replying fast. I have done the same thing.Actually  cordova push notification received event is not getting triggered only when app is in background.How I will know that playload is foreground or coldstart when my notification received function is not getting called.
I am using parse push notification service so I dont have  sender Id so I am using client key as the sender Id is that creating problem.But if that was the case I wouldn't have recive notification when app is running right.
So can please give me accurate solution.

**Comment by GopalakrishnanK** [comment]
what about having more than 1 notifications...?...Only the last received notification will work right? Any ideas...?

**Comment by madsongr** [comment]
@oscarabilleira Hi man, how do you send the id? If you use the script below, do you include a new array item inside $msg?

[https://phonegappro.com/tutorials/apache-cordova-phonegap-push-notification-tutorial-part-3/](url)

**Comment by GaneshSawant** [comment]
Hello Everyone, 

I am using cordova based forcedroid sdk package to build hybrid remote app.
I have added onResume event inside my landing page(Its a Salesforce VF page) for various purposes like Session refresh and logout functionality.

Can someone please help me understand, how to check if the OnResume event triggered because of Push notification tapping ? 
Not sure how to access Push payload on Salesforce visualforce page as mentioned above.

Thank you so much for your help :)

**Comment by musasheikh** [comment]
hi @oscarabilleira 

Only the last received notification will work, do you have any other solution plz.
